### PR TITLE
Added documentation, Stubbed out localization, Fixed observers

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
Discovered while documenting BridgeClientUI that the AppConfig, Study, and UserSessionInfo all require 2x kill and relaunch to get updates. There doesn't appear to be anything that triggers updates to these objects proactively (ie. you have to restart once to refresh), but it will get updates w/ first app restart.

This also includes a bunch of documentation changes and I stubbed out localization.

### CODE REVIEW

File: SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift [hunk 1]
```diff
@@ -35,32 +35,46 @@ import Research
...
+    /// Pointer to the app manager singleton for this application.
     public var bridgeManager: BridgeClientAppManager!
     
     private var _defaultTaskToSchemaMapping: [String : String]?
     private var _appConfigTaskToSchemaMapping: [String : String]?
     
+    /// Set up the archive manager by attaching the ``BridgeClientAppManager`` singleton to it.
+    ///
+    /// - Parameters:
+    ///   - bridgeManager: The app manager singleton for this application
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift]:
```suggestion
    ///   - bridgeManager: The app manager singleton for this application.
```

File: SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift [hunk 2]
```diff
@@ -246,19 +260,41 @@ extension AppConfig {
...
+/// presented to the participant as a single "task" that then needs to be split into different archives for
+/// post-processing. This structure of having a manager for the state object and a separate class for holding
+/// the archive data is intended to allow for that scenario.
 open class SageResearchResultArchive : AbstractResultArchive, RSDDataArchive {
     
     /// Hold the task result (if any) used to create the archive.
     internal var taskResult: RSDTaskResult?
     
+    /// Whether or not the archive should include "reserved" files that are collated separately from the
+    /// default behavoir of iteratively looking through the result for objects that support the ``RSDArchivable``
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift]:
```suggestion
    /// default behavior of iteratively looking through the result for objects that support the ``RSDArchivable``
```

File: SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift [hunk 3]
```diff
@@ -246,19 +260,41 @@ extension AppConfig {
...
+    /// - Returns: Whether or not the file should be added. (Default == true)
     open func shouldInsertData(for filename: RSDReservedFilename) -> Bool {
         true
     }
     
+    /// The archivable object to use to process the given result.
+    ///
+    /// - Parameters:
+    ///   - result: The result to process.
+    ///   - sectionIdentifier: The section identifier for result within the assessment.
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift]:
```suggestion
    ///   - sectionIdentifier: The section identifier for the result within the assessment.
```

File: SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift [hunk 4]
```diff
@@ -246,19 +260,41 @@ extension AppConfig {
...
+    ///   - stepPath: The step path for the result within the assessment
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Data Archiving/SageResearchArchiveManager.swift]:
```suggestion
    ///   - stepPath: The step path for the result within the assessment.
```

File: SwiftPackage/Sources/BridgeClientUI/Extensions/Date+Utils.swift [hunk 5]
```diff
@@ -36,18 +36,27 @@ import BridgeClient
 
 extension Date {
     
+    /// Convenience method for returning the "start" of the day (ie, midnight).
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Extensions/Date+Utils.swift]:
```suggestion
    /// Convenience method for returning the "start" of the day (i.e., midnight).
```

File: SwiftPackage/Sources/BridgeClientUI/Extensions/Date+Utils.swift [hunk 6]
```diff
@@ -36,18 +36,27 @@ import BridgeClient
...
+    ///
+    /// - Returns: The start of the day using the current  calendar.
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Extensions/Date+Utils.swift]:
```suggestion
    /// - Returns: The start of the day using the current calendar.
```

File: SwiftPackage/Sources/BridgeClientUI/Extensions/Date+Utils.swift [hunk 7]
```diff
@@ -36,18 +36,27 @@ import BridgeClient
...
     public var isToday: Bool {
         Calendar.current.isDateInToday(self)
     }
     
+    /// Is this date "tomorrow"?
     public var isTomorrow: Bool {
         Calendar.current.isDateInTomorrow(self)
     }
     
+    /// This date localized using the participants preferred locale.
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Extensions/Date+Utils.swift]:
```suggestion
    /// This date localized using the participant's preferred locale.
```

File: SwiftPackage/Sources/BridgeClientUI/LocalNotificationManager.swift [hunk 8]
```diff
@@ -37,14 +37,24 @@ import BridgeClient
 fileprivate let kAllowSnoozeKey = "allowSnooze"
 fileprivate let maxTotalNotifications = 60
 
+/// A manager that be used to set up and respond to local notifications. The base class will set up notifications
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/LocalNotificationManager.swift]:
```suggestion
/// A manager that is used to set up and respond to local notifications. The base class will set up notifications
```

File: SwiftPackage/Sources/BridgeClientUI/Models/AssessmentInfoMap.swift [hunk 9]
```diff
@@ -34,13 +34,32 @@ import SwiftUI
...
+    /// The color to use when displaying an `AssessmentTimelineCardView`.
     func color() -> Color
 }
 
+/// The `AssessmentInfoMap` is an environmental variable that is used to allow shared views
+/// to know about the `Color`, `Text`, and `Image` to use for a given assessment while
+/// allowing those values to be set at the app level. Since this framework does not directly link in
+/// any assessment libraries, the app has the responsibility for setting up those mappings.
+///
+/// - Note: This struct uses `BridgeClient.AssessmentInfo` in it's mapping and its
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Models/AssessmentInfoMap.swift]:
```suggestion
/// - Note: This struct uses `BridgeClient.AssessmentInfo` in its mapping and its
```

File: SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+    @Published public var phase: Phase?
+    
+    /// The type of sign in used by this study.
+    @Published public var signInTypes: [SignInType]?
+
+    /// An in-depth description of this study for prospective participants, or to show participants after
+    /// they have joined a study (the “about” screen for the study).
+    @Published public var details: String?
+    
+    /// An identification number for the study that has been assigned by the sponsor or institution running the study.
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift:59]: If it's a number, is there a particular reason the type is String?

File: SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift [hunk 11]
```diff
@@ -0,0 +1,320 @@
...
+public struct IRBProtocolInfo : Hashable, Identifiable {
+    public var id: Int { self.hashValue }
+    
+    /// The primary IRB contact for this study.
+    public let contact: StudyContact
+    
+    /// The name of the protocol as it was submitted to the IRB for approval.
+    public let protocolName: String?
+    
+    /// An identification number for the study that has been assigned by the IRB.
```
Comment (Erin-Mounts) [inline SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift:202]: Again, if it's a number, why is the type String?

**Review by Erin-Mounts — APPROVED** [review]
A few nits and a couple of questions, otherwise looks good.

File: SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift (hunk 10 above)
Comment (syoung-smallwisdom) [reply SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift:59]: I have no idea. The description is a copy/paste from Swagger.

File: SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift (hunk 11 above)
Comment (syoung-smallwisdom) [reply SwiftPackage/Sources/BridgeClientUI/Models/StudyObserver.swift:202]: Also a copy/paste from Bridge model

**Pull request merged**